pennylane/ops/op_math/controlled.py

```diff
@@ -204,6 +224,62 @@ def wrapper(*args, **kwargs):
     return wrapper
 
 
+@functools.lru_cache  # only create the first time requested
+def _get_ctrl_qfunc_prim():
+    """See capture/explanations.md : Higher Order primitives for more information on this code."""
+    # if capture is enabled, jax should be installed
+    import jax  # pylint: disable=import-outside-toplevel
+
+    ctrl_prim = jax.core.Primitive("ctrl_transform")
+    ctrl_prim.multiple_results = True
+
+    @ctrl_prim.def_impl
+    def _(*args, n_control, jaxpr, control_values, work_wires, n_consts):
+        consts = args[:n_consts]
+        args = args[n_consts:]
+        control_wires = args[-n_control:]
+        args = args[:-n_control]
```

Comment:
This is a very subjective suggestion, but in my head this reads easier:
```suggestion
        consts = args[:n_consts]
        control_wires = args[-n_control:]
        args = args[n_consts:-n_control]
```